### PR TITLE
feat(AUTH-CPC-1189): users can reset their password

### PR DESCRIPTION
**Pull request by flyindonut**
**JIRA:** https://champlainsaintlambert.atlassian.net/browse/CPC-1189
## Context:
Allows users to reset their password in case they forgot it.
## Does this PR change the .vscode folder in petclinic-frontend?:
Nope
## Changes

- Created v2 endpoints for forgot and reset password
- Modified processForgotPassword in UserServiceImpl to allow different URLs
- Added url field in UserEmailRequestDTO
- Created new pages for forgot password and reset password in the front end as well as their corresponding API calls, components, and models.
- Routed the pages properly
- Improved the axiosInstance and the axiosErrorResponseHandler (Errors are now handled properly and can be intercepted in a try-catch for customizable error handling on a page)
- Added a forgot password link in the login page
- Created all the necessary tests to cover the implementation of the new endpoints
- Fixed Navbar that https://github.com/cgerard321/champlain_petclinic/pull/867 modified :disappointed: 

## Before and After UI (Required for UI-impacting PRs)
Before:
Login Page
image
After:
Login Page
image
Forgot Password Page
image
image
image
Reset Password Page
image
image
image
image

## Linked PR
https://github.com/cgerard321/champlain_petclinic/pull/867

### CODE REVIEW

File: petclinic-frontend/src/features/bills/AdminBillsListTable.tsx [hunk 1]
```diff
@@ -315,7 +315,7 @@ export default function AdminBillsListTable(): JSX.Element {
           </p>
         </div>
       ) : (
-        <div className="container">
+        <div className="admin-bills-list-table-container">
```
Comment (Justin222993) [inline petclinic-frontend/src/features/bills/AdminBillsListTable.tsx:318]: Good job changing it to a bill-specific class so it doesn't mess with other container class such as the navbar :)

File: petclinic-frontend/src/shared/api/axiosInstance.ts [hunk 2]
```diff
@@ -1,29 +1,42 @@
...
-      // return Promise.reject(response); // this is temporary until we have a our global axios error handler that redirects to the correct pages for the responses.
-    } else {
-      return Promise.reject(error as Error);
+  // response interceptor to handle errors globally
+  instance.interceptors.response.use(
+    response => response,
+    error => {
+      // handle errors through a custom error handler
+      handleAxiosError(error);
+      // yes, now try-catch can actually catch the error unlike before :D
```
Comment (Justin222993) [inline petclinic-frontend/src/shared/api/axiosInstance.ts:20]: Good catch!

**Review by Justin222993 — APPROVED** [review]
Great work! Since you modified something in the bill CSS, you should add someone on their team for them to approve it as well.

File: petclinic-frontend/src/features/bills/AdminBillsListTable.tsx (hunk 1 above)
Comment (Futurespast) [reply petclinic-frontend/src/features/bills/AdminBillsListTable.tsx:318]: thank you for catching that for us!

File: petclinic-frontend/src/features/bills/AdminBillsListTable.css [hunk 3]
```diff
@@ -1,4 +1,4 @@
-.container {
+.admin-bills-list-table-container {
```
Comment (Futurespast) [inline petclinic-frontend/src/features/bills/AdminBillsListTable.css:1]: Thanks for catching that for us

**Review by Futurespast — APPROVED** [review]

**Pull request merged**